TESTING.md

```diff
@@ -0,0 +1,74 @@
+
+# Requirements
+
+In order to run the full test suite users will need access to a running Manta
+endpoint. Some tests rely on additional configuration which must occur outside
+of this project and will be skipped if it is determined that the test cannot
+be run. Keep an eye on the count of skipped tests and review the relevant
+sections in this document if you find that any tests are skipped.
+
+## Client Configuration
+
+The following properties should be set using either environment variables or
+system properties:
+
+ - `manta.url` or `MANTA_URL`
+ - `manta.user` or `MANTA_USER`
+ - `manta.key_id` or `MANTA_KEY_ID`
+ - `manta.key_path` or `MANTA_KEY_PATH`
+
+## Multipart Upload (Server-side support)
+Tests specific to server-side MPU functionality will be automatically skipped
+if the server con does not support that capability.
+
+## Roles
+Tests which rely on Triton roles will require the creation of roles through the
+Triton customer portal. The names of the roles used in the tests will be
+retrieved in the same fashion as standard configuration, first checking
+environment variables followed by system properties and finally expecting
+defaults to be defined. These tests will be skipped if it is determined that the
+roles do not exist.
+
+| Default     | System Property           | Environment Variable      |
+|------------ | ------------------------- | --------------------------|
+| primary     | manta.it.role_primary     | MANTA_IT_ROLE_PRIMARY     |
+| secondary   | manta.it.role_secondary   | MANTA_IT_ROLE_SECONDARY   |
+
+## Integration Test Path
+Integration tests only touch paths under the integration test root path which
+is configurable using the `manta.it.path` system property or the `MANTA_IT_PATH`
+environment variable. The default behavior is to compute the base path based on
+the user's private folder, e.g. `/my.account/stor/java-manta-integration-tests`.
+
+# Running Tests
+
+Run `mvn verify` from the project root to run all tests. Some Maven goals will
```

Comment:
I'm not sure what the statement that some goals don't work is referring to.

Reply:
I don't remember. We need to reverify, but I do remember some goals don't work when manually executed.

Reply:
`mvn validate` for example:

`Could not resolve following dependencies: [com.joyent.manta:java-manta-benchmark:jar:3.1.7-SNAPSHOT (test)]`